pkg/controllers/disruption/controller.go

```diff
@@ -234,6 +229,27 @@ func (c *Controller) executeCommand(ctx context.Context, m Method, cmd Command,
 		c.cluster.UnmarkForDeletion(providerIDs...)
 		return fmt.Errorf("adding command to queue (command-id: %s), %w", commandID, multierr.Append(err, state.RequireNoScheduleTaint(ctx, c.kubeClient, false, stateNodes...)))
 	}
+
+	// An action is only performed and pods/nodes are only disrupted after a successful add to the queue
+	ActionsPerformedCounter.With(map[string]string{
+		actionLabel:            string(cmd.Action()),
+		methodLabel:            m.Type(),
+		consolidationTypeLabel: m.ConsolidationType(),
+	}).Inc()
+	for _, cd := range cmd.candidates {
+		NodesDisruptedCounter.With(map[string]string{
+			metrics.NodePoolLabel:  cd.nodePool.Name,
+			actionLabel:            string(cmd.Action()),
+			methodLabel:            m.Type(),
+			consolidationTypeLabel: m.ConsolidationType(),
+		}).Inc()
+		PodsDisruptedCounter.With(map[string]string{
+			metrics.NodePoolLabel:  cd.nodePool.Name,
+			actionLabel:            string(cmd.Action()),
+			methodLabel:            m.Type(),
+			consolidationTypeLabel: m.ConsolidationType(),
+		}).Add(float64(len(cd.reschedulablePods)))
+	}
```

Comment:
**Shallow Comment**

Seems like iterating through candidates here is expensive for metric. Is there somewhere we can do this inside of the waitOrTerminate function in the orchestration queue?

Reply:
After checking the code, looks cleaner to separate this out after the fact.

Reply:
Yeah, I think this is a reasonable place to do it, personally